processor/servicegraphprocessor/processor_test.go

```diff
@@ -98,33 +100,61 @@ func TestProcessorShutdown(t *testing.T) {
 }
 
 func TestProcessorConsume(t *testing.T) {
-	// Prepare
-	cfg := &Config{
-		MetricsExporter: "mock",
-		Dimensions:      []string{"some-attribute", "non-existing-attribute"},
-	}
-
-	mockMetricsExporter := newMockMetricsExporter(func(md pmetric.Metrics) error {
+	metricsExporter := newMockMetricsExporter(func(md pmetric.Metrics) error {
 		return verifyMetrics(t, md)
 	})
 
-	processor := newProcessor(zaptest.NewLogger(t), cfg, consumertest.NewNop())
+	_ = featuregate.GlobalRegistry().Set(virtualNodeFeatureGate.ID(), true)
```

Comment:
Should you unset this when the test finishes?

Reply:
It doesn't matter if we don't reset it, but I'll reset it anyway